packages/python/plotly/plotly/express/_chart_types.py

```diff
@@ -357,6 +358,53 @@ def bar(
 bar.__doc__ = make_docstring(bar, append_dict=_cartesian_append_dict)
 
 
+def timeline(
+    data_frame=None,
+    x_start=None,
```

Comment:
if these arguments are required they should not be keywords arguments but positional ones I think?

Reply:
we could think of defaults like providing a global `start` or `end` parameter which would be the one for all. But I think the easiest solution is to make the arguments positional.

Reply:
I would really like to keep this function consistent with all the others, which have the `data_frame` argument come first. I agree that in general required arguments should come first and be positional, but in this case I prefer consistency. `px.pie()` sort of works the same way: you need to provide either `values` or `names` otherwise you get an empty figure.

To align with something like `pie` I could just have `x` be unbound if `x_start` and `x_end` aren't provided instead of raising (i.e. we would render an empty figure just like `px.pie(df)`) but I think this would be less helpful for users.

Reply:
I see your point but preferring px consistency over python usage (when I see a keyword argument I don't expect the function to raise with its default value, and in particular None usually means optional) is not 100% satisfying. You are right that pie has the same problem, and that if you use all the default values for the other functions, the result is an empty figure so not very interesting. I'd be happy with a modification of the docstring for `x_start` and `x_end` mentioning that they are compulsory.

Reply:
> To align with something like `pie` I could just have `x` be unbound if `x_start` and `x_end` aren't provided instead of raising (i.e. we would render an empty figure just like `px.pie(df)`) but I think this would be less helpful for users.

but this would be consistent :-)